WebContent: Switch scroll direction on shift modifier

can3p opened the pull request:
This pull request changes web content component behavior to be in line with other scrollable areas - vertical mouse scroll turns horizontal in case shift modifier key is pressed.

To be honest, I'm not entirely sure I've put it on the right level, since I'm still yet to get familiar with the whole ipc / components structure and interactions. Will change if there is a better place

alimpfard reviewed (commented):
I'd suggest at least moving this down to `EventHandler::handle_mousewheel`.
Also, maybe consider swapping the deltas instead of setting the `y` component to zero (bonus: less code :^)

can3p commented:
Thanks @alimpfard ! I've moved the code and changed it to swapping. Not sure if it's really needed, since shift is used in conjunction with vertical scroll, but that indeed removes a special case. Anyway, I'm fine with this approach too :)

IdanHo commented on Userland/Libraries/LibWeb/Page/EventHandler.cpp:
```diff
@@ -117,6 +117,12 @@ bool EventHandler::handle_mousewheel(const Gfx::IntPoint& position, unsigned int
     if (!layout_root())
         return false;
 
+    if (modifiers & KeyModifier::Mod_Shift) {
+        auto temp = wheel_delta_x;
+        wheel_delta_x = wheel_delta_y;
+        wheel_delta_y = temp;
+    }
```
```suggestion
    if (modifiers & KeyModifier::Mod_Shift)
        swap(wheel_delta_x, wheel_delta_y);
```
